src/chronify/time_series_mapper_index_time.py

```diff
@@ -307,7 +316,121 @@ def _create_interm_map_with_time_zone_and_dst_adj_drop_dup(
 
         mapping_schema = MappingTableSchema(
-            name="mapping_table",
+            name="index_time_mapping_table_duplicate_fallback",
+            time_configs=[from_time_config, mapped_schema.time_config],
+        )
+        return df, mapping_schema, mapped_schema
+
+    def _create_interm_map_with_tz_and_dst_adj_drop_interpolate(
+        self,
+    ) -> tuple[pd.DataFrame, MappingTableSchema, TableSchema]:
+        """Create mapping dataframe for converting INDEX_LOCAL time to its represented datetime
+        with time-based daylight_saving adjustment that
+        drops the spring-forward hour and interpolates the fall-back hour
+        """
+        mapped_schema = self._create_intermediate_schema()
+        assert isinstance(mapped_schema.time_config, DatetimeRange)
+        mapped_time_col = mapped_schema.time_config.time_column
+        mapped_time_data_ntz = make_time_range_generator(
+            mapped_schema.time_config
+        ).list_timestamps()
+
+        from_time_col = "from_" + self._from_time_config.time_column
+        from_time_data = make_time_range_generator(self._from_time_config).list_timestamps()
+
+        df_ntz = pd.DataFrame(
+            {
+                from_time_col: from_time_data,
+                "clock_time": mapped_time_data_ntz,
+            }
+        )
+        df_ntz["clock_time"] = df_ntz["clock_time"].astype(str)
+
+        tz_col = self._from_time_config.get_time_zone_column()
+        assert tz_col is not None, "Expecting a time zone column for INDEX_LOCAL"
+        from_tz_col = "from_" + tz_col
+
+        with self._engine.connect() as conn:
+            table = Table(self._from_schema.name, self._metadata)
+            stmt = select(table.c[tz_col]).distinct().where(table.c[tz_col].is_not(None))
+            time_zones = read_database(stmt, conn, self._from_time_config)[tz_col].to_list()
+
+        from_time_config = self._from_time_config.model_copy()
+        assert isinstance(from_time_config, IndexTimeRangeLocalTime)
+        from_time_config.time_column = from_time_col
+        from_time_config.time_zone_column = from_tz_col
+
+        df_tz = []
+        to_tz = self._to_time_config.start.tzinfo
+
+        limit = timedelta(hours=1) / self._from_time_config.resolution
+        assert (limit % 1 == 0) and (limit > 0), f"limit must be an integer, {limit}"
+        limit = int(limit)
+
+        for time_zone in time_zones:
+            config_tz = self._create_local_time_config(time_zone)
+            time_data = make_time_range_generator(config_tz).list_timestamps()
+            # Extract clock time
+            clock_time_data = [x.strftime("%Y-%m-%d %H:%M:%S") for x in time_data]
+            # Preemptively convert to dst time tzinfo, otherwise pandas treats the col,
+            # which consists of the timeseries of different time zones, as an object col
+            mapped_time_data = [x.tz_convert(to_tz) for x in time_data]
+            df_map = pd.DataFrame(
```

Comment:
This is the start of the section that's different from `_create_interm_map_with_tz_and_dst_adj_drop_duplicate` and it is leveraging similar logic from the time resolution mapping work.